apps/api-extractor/src/collector/Collector.ts

```diff
@@ -511,37 +511,16 @@ export class Collector {
       this._calculateMetadataForDeclaration(astDeclaration);
     }
 
-    // Do any of the declarations have a release tag?
+    // The most public effectiveReleaseTag for all declarations
     let maxEffectiveReleaseTag: ReleaseTag = ReleaseTag.None;
 
     for (const astDeclaration of astSymbol.astDeclarations) {
       // We know we solved this above
       const declarationMetadata: DeclarationMetadata = astDeclaration.metadata as DeclarationMetadata;
       const effectiveReleaseTag: ReleaseTag = declarationMetadata.effectiveReleaseTag;
 
-      if (effectiveReleaseTag !== ReleaseTag.None) {
-        if (maxEffectiveReleaseTag !== ReleaseTag.None && maxEffectiveReleaseTag !== effectiveReleaseTag) {
-          if (!astSymbol.isExternal) { // for now, don't report errors for external code
-            switch (astDeclaration.declaration.kind) {
-              case ts.SyntaxKind.FunctionDeclaration:
-              case ts.SyntaxKind.MethodDeclaration:
-                // For function and method overloads, take the highest release from multiple declarations
-                // TODO: Expand this out to more kinds
-                if (maxEffectiveReleaseTag < effectiveReleaseTag) {
-                  maxEffectiveReleaseTag = effectiveReleaseTag;
-                }
-                break;
-              default:
-                this.messageRouter.addAnalyzerIssue(
-                  ExtractorMessageId.DifferentReleaseTags,
-                  'This symbol has another declaration with a different release tag',
```

Comment:
I've moved this logic to the `ValidationEnhancer`, to avoid the complexity of a partially constructed `SymbolMetadata`.  I originally put it here in the `Collector` thinking that if an error was suppressed, we might need to correct the `effectiveReleaseTag` while reporting the error.  But so far that need hasn't arisen.